navigation_and_routing/lib/src/routing/route_state.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'parsed_route.dart';
+import 'parser.dart';
+
+/// The current route state. To change the current route, call obtain the state using
+/// `RouteState.of(context)` and call `go()`:
```

Comment:
RouteStateScope instead of RouteState, here and below? (or move the implementation of `of`)?